packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -181,6 +222,7 @@ class DayPicker extends StatelessWidget {
     @required this.firstDate,
     @required this.lastDate,
     @required this.displayedMonth,
+    this.onHeaderTap,
```

Comment:
It's not clear to me why the headerTap callbacks were added? They don't seem be part of addressing the original issue.